packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -190,7 +190,18 @@ Future<void> _runBuild(
 
   // MSBuild sends all output to stdout, including build errors. This surfaces
   // known error patterns.
-  final RegExp errorMatcher = RegExp(r':\s*(?:warning|(?:fatal )?error).*?:');
+  final List<String> errorMessages = <String>[
+    r':\s*(?:warning|(?:fatal )?error).*?:',
+    r'Error detected in pubspec\.yaml:',
+  ];
+  // Known secondary error lines for pubspec.yaml errors:
+  final List<String> pubspecErrorDetails = <String>[
+    r'No file or variants found for asset:',
+  ];
+  final RegExp errorMatcher = RegExp(<String>[
+    ...errorMessages,
+    ...pubspecErrorDetails,
+  ].join('|'));
```

Comment:
Let's keep this simple, the string joining makes readability a tiny bit harder:

```suggestion
  final RegExp errorMatcher = RegExp(
    r'(:\s*(?:warning|(?:fatal )?error).*?:)|'
    r'Error detected in pubspec\.yaml:|'
    r'No file or variants found for asset:'
  );
```

Reply:
Hm. What do you think of something like this?

```dart
  final RegExp errorMatcher = RegExp(
	<String>[
      // Known error messages
      r'(:\s*(?:warning|(?:fatal )?error).*?:)',
      r'Error detected in pubspec\.yaml:',
    
      // Known secondary error lines for pubspec.yaml
      r'|No file or variants found for asset:',
    ].join('|'),
  );
```

Reply:
That seems fine to me (other than the extra inline `|` which I assume is a typo).

Reply:
Yup that was a typo, I updated the code snippet

Reply:
I will test the new suggetion locally and commit. Thanks.